### PR TITLE
Add support for application/smile

### PR DESCRIPTION
**Pull request by dblock**
### Description

Add support for application/smile.

Thanks to @ngyewch for the recent work on the smile-js library to get this over the line!

### Issues Resolved

Closes https://github.com/opensearch-project/opensearch-api-specification/issues/372.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tests/cat/health.yaml [hunk 1]
```diff
@@ -82,3 +82,16 @@ chapters:
...
           node.data: '1'
           discovered_cluster_manager: 'true'
+  - synopsis: Cat in different formats (format=smile).
+    method: GET
+    path: /_cat/health
+    parameters:
+      format: smile
+    response:
+      status: 200
+      content_type: application/yaml
```
Comment (Xtansia) [inline tests/cat/health.yaml:92]: Are we not actually validating `content_type` against the response content-type header?

File: tools/tests/tester/fixtures/stories/passed.yaml [hunk 2]
```diff
@@ -44,3 +44,13 @@ chapters:
...
       payload:
         - node.total: '1'
+  - synopsis: This GET /_cat/health chapter returns application/smile and should pass.
+    path: /_cat/health
+    parameters:
+      format: cbor
+    method: GET
+    response:
+      status: 200
+      content_type: application/smile
```
Comment (Xtansia) [inline tools/tests/tester/fixtures/stories/passed.yaml:54]: As above regarding validating `content_type` as test is supposed to be for `smile` but format is set to `cbor`

**Review by Xtansia — CHANGES_REQUESTED** [review]

File: tests/cat/health.yaml (hunk 1 above)
Comment (dblock) [reply tests/cat/health.yaml]: Looks like it, https://github.com/opensearch-project/opensearch-api-specification/issues/392.

File: tools/tests/tester/fixtures/stories/passed.yaml (hunk 2 above)
Comment (dblock) [reply tools/tests/tester/fixtures/stories/passed.yaml]: Updated to be `format: smile`.

**Review by Xtansia — APPROVED** [review]

**Pull request merged**